draft-irtf-cfrg-opaque.md

```diff
@@ -1392,33 +1393,45 @@ Output:
 
 Steps:
 1. dh1 = SerializePublicKey(sk1 * pk1)
-2. dh2 = SerializePublicKey(sk2 * pk2)
-3. dh3 = SerializePublicKey(sk3 * pk3)
-4. Output concat(dh1, dh2, dh3)
+1. dh2 = SerializePublicKey(sk2 * pk2)
+1. dh3 = SerializePublicKey(sk3 * pk3)
+1. Output concat(dh1, dh2, dh3)
 ~~~
 
-Using this shared secret, further keys used for encryption and authentication are
-computed using the following function.
-
 ~~~
-DeriveKeys(ikm, preamble)
+DeriveKeys(role, preamble,
+            secret_keyshare, peer_keyshare, private_key, peer_public_key)
 
 Input:
-- ikm, input key material.
-- preamble, the transcript as defined by Preamble().
+- role, either "client" or "server".
+- preamble, the protocol transcript with identities and messages.
+- secret_keyshare, the role's secret keyshare.
+- peer_keyshare, the peer's public keyshare
+- private_key, the role's private key.
+- peer_public_key, the peer's public key.
 
 Output:
-- Km2, a MAC authentication key.
-- Km3, a MAC authentication key.
 - session_key, the shared session secret.
+- server_mac, the server's MAC.
+- client_mac, the client's MAC.
 
 Steps:
+1. if role == "client",
```

Comment:
I don't really like this -- can we just pass `ikm` in as input?

Reply:
yeah, probably